--- menu.c	2010-09-08 22:09:39.479182000 -0400
+++ menufull.c	2010-09-08 22:09:32.101996000 -0400
@@ -197,6 +197,7 @@
 void
 drawHiscores(int max)
 {
+	//int i;
 	//SDL_Rect a,b;
 
 	/* erase the screen */
@@ -216,6 +217,15 @@
 
 	writeCString(gfx, screen, 65, 125, "requires full version", 1);
 
+	/*for(i=0;i<max;i++) {
+		writeNumber(gfx, screen, 10, 23+i*17, i+1, 2);
+		writeCString(gfx, screen, 30, 20+i*17, hiscore[i].name, 0);
+		writeCString(gfx, screen, 180, 20+i*17, "st",0);
+		writeNumber(gfx, screen, 200, 23+i*17, hiscore[i].stage,2);
+		writeCString(gfx, screen, 236, 20+i*17, "sc",0);
+		writeNumber(gfx, screen, 260, 23+i*17, hiscore[i].score,6);
+	}*/
+
 	SDL_Flip(screen);
 }
 
@@ -224,8 +234,13 @@
 {
 	Uint32 tick;
 	SDL_Event mevent;
+	//int i;
 	
 	drawHiscores(1);
+	/*for(i=0;i<10;i++) {
+		drawHiscores(i+1);
+		SDL_Delay(300);
+	}*/
 
 	tick=SDL_GetTicks();
 	while(1) {
@@ -291,7 +306,7 @@
 	} 
 	else {
 		writeCString(gfx, screen, 20, 90,  "   accelerometer", (option==1));
-		writeCString(gfx, screen, 70, 125, "requires full version",1);		
+		writeCString(gfx, screen, 70, 125, "tilt to control",1);		
 		}
 	writeCString(gfx, screen, 80, 165, "tap screen to fire",1);	
 
@@ -343,9 +358,10 @@
 						default:
 						break;
 						case 1:
+							if(joy[0]) {
 								conf.control[0]=conf.control[0] ? 0 : 1;
 								drawConfigure(option);
-							
+							}
 						break;
 						case 2://3
 							conf.sound--;
